src/views/ApartmentApp/components/StudentApplication/components/OffCampusSection/components/ProgramListItem/index.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect, useRef } from 'react';
+import isEqual from 'lodash/isEqual';
+import {
+  Grid,
+  Divider,
+  ListItem,
+  ListItemSecondaryAction,
+  MenuItem,
+  FormControl,
+  Input,
+  InputLabel,
+  Select,
+  IconButton,
+} from '@material-ui/core';
+import ClearIcon from '@material-ui/icons/Clear';
+
+const ProgramListItem = ({
+  disabled,
+  index,
+  availableMajors,
+  availableApplicants,
+  onOffCampusChanged,
+  onOffCampusApplicantRemove,
+}) => {
+  const [applicantMemberValue, setMemberValue] = useState(''); // Applicant drop-down menu value
+  const [applicantDepartmentValue, setDepartmentValue] = useState(''); // Major drop-down menu value
+
+  useEffect(() => {
+    if (isEqual(previousInputs.current, [index, availableApplicants])) {
+      return;
+    }
+    const getMembersFromProps = () => {
+      setMemberValue(availableApplicants[index].name);
+      setDepartmentValue(availableApplicants[index].major);
+    };
+
+    getMembersFromProps();
+  });
+
+  const previousInputs = useRef();
+  useEffect(() => {
+    previousInputs.current = [index, availableApplicants];
+  });
```

Comment:
```suggestion
const ProgramListItem = ({
  disabled,
  index,
  applicantName,
  departmentName,
  availableDepartments,
  onOffCampusChanged,
}) => {
  const [applicantDepartmentValue, setDepartmentValue] = useState(''); // Major drop-down menu value

  useEffect(() => {
    setDepartmentValue(departmentName);
  }, [departmentName);
```
If you eliminate the dropdown of the applicants and just have it be a fixed map of the applicants already in the list, then you can use the applicant info directly from the props. Additionally, it is more reliable to pass the applicant and department values directly in props rather than trying to get them out of the applicants array, thus eliminating the complexity of trying the use the isEqual to perform a deep comparison, which was actually a flawed workaround that I tried to use before I figured out what was actually going wrong with the hall array